boskos/metrics/metrics.go

```diff
@@ -30,62 +32,57 @@ import (
 )
 
 type prometheusMetrics struct {
-	GceStats map[string]prometheus.Gauge
-	GkeStats map[string]prometheus.Gauge
+	// map[resource-name]map[state]Gauge
+	BoskosState map[string]map[string]prometheus.Gauge
 }
 
 var (
 	promMetrics = prometheusMetrics{
-		GceStats: map[string]prometheus.Gauge{
-			common.Free: prometheus.NewGauge(prometheus.GaugeOpts{
-				Name: "boskos_gce_project_free",
-				Help: "Number of free gce-project",
-			}),
-			common.Busy: prometheus.NewGauge(prometheus.GaugeOpts{
-				Name: "boskos_gce_project_busy",
-				Help: "Number of busy gce-project",
-			}),
-			common.Dirty: prometheus.NewGauge(prometheus.GaugeOpts{
-				Name: "boskos_gce_project_dirty",
-				Help: "Number of dirty gce-project",
-			}),
-			common.Cleaning: prometheus.NewGauge(prometheus.GaugeOpts{
-				Name: "boskos_gce_project_cleaning",
-				Help: "Number of cleaning gce-project",
-			}),
-		},
-		GkeStats: map[string]prometheus.Gauge{
+		BoskosState: map[string]map[string]prometheus.Gauge{},
+	}
+
+	// I need to access to the list of projects in init, so I cannot do a flag :-\
+	resources = []string{
+		"gce-project",
```

Comment:
It should not be hard coded! This is not usable for istio.

Reply:
ok, I guess I'll look into a way to access flags in init? or I can put that into an env var?

Reply:
fixed